cs/di-configuration.texy

```diff
@@ -175,12 +175,63 @@ public function createServiceArticles()
 }
 \--
 
-Nad odkazovanou službou lze volat metody, nicméně pro jednoduchost místo `->` používáme `::`.
+
+Anonymní služby
+---------------
+
+Pojmenování služeb je vhodné hlavně ve chvíli, kdy na ně chceme odkazovat z jiných částí konfiguračního souboru. Pokud na službu není odkazováno názvem, není ji potřeba pojmenovávat. Pro anonymní služby použijte následující syntax:
 
 /--neon
 services:
-	routerFactory: App\Router\Factory(@http.request::getMethod())  # http.request je systémová služba
-	router: @routerFactory::create()
+	- PDO('sqlite::memory:)
+
+	-
+		factory: Model\ArticleRepository
+		setup:
+			- setCacheStorage
+\--
+
+Nicméně i na anonymní služby lze odkazovat, jen místo názvu uvedeme jejich typ (třídu nebo rozhraní):
+
+/--neon
+services:
+	router: @App\Router\Factory::create
+\--
+
+Službu získáme z DI kontejneru metodou `getByType()`:
+
+/--php
+$database = $container->getByType(PDO::class);
+\--
+
+
+
+Pokročilá syntaxe
+-----------------
+
+Volání method lze pohodlně zřetězit za sebe pomocí zápisu s dvěma dvojtečkami (`::`).
+
+/--neon
+services:
+	foo: FooFactory::create()::last()
+\--
+
+Vygeneruje:
+
+/--php
+public function createServiceFoo()
+{
+	return FooFactory::create()->another()->last();
+}
+\--
+
+Nad odkazovanou službou lze volat taktéž metody, opět používáme zápis s dvojtečkami (`::`).
+
+/--neon
+routerFactory: 
+	factory: App\Router\Factory
+	setup:
+		- setMethod(@http.request::getMethod())
```

Comment:
Tady je potřeba upravit co to vygeneruje.

```
public function createServiceRouterFactory()
{	
	$service = new App\Router\Factory;
    $service->setMethod( $this->getService('http.request')->getMethod() );
    return $service;
}	
```

Reply:
Promin, prehledl jsem, opraveno.